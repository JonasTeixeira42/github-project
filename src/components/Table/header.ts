export default ['Id', 'Repository', 'URL']
