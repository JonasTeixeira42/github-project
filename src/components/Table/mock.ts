import { Repositories } from 'hooks/use-repositories'

const repositories: Repositories[] = [
  {
    id: 1,
    name: 'Repo 1',
    html_url: 'https://www.google.com/'
  }
]

export default repositories
