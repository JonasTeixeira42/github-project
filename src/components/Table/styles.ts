import styled, { css } from 'styled-components'

export const Wrapper = styled.div`
  ${({ theme }) => css`
    border: 0.1rem solid ${theme.colors.lightGray};
    border-radius: ${theme.border.radius};
    padding: ${theme.spacings.xxsmall};
  `}
`

export const Row = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: repeat(3, 1fr);

    &:not(:last-child) {
      border-bottom: 0.1rem solid ${theme.colors.lightGray};
      padding-bottom: 0.8rem;
      margin-bottom: 0.8rem;
    }
  `}
`

export const Cell = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.xsmall};
    padding: 0 0.5rem;
  `}
`

export const Link = styled.a``

export const Header = styled.h2`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
  `}
`
