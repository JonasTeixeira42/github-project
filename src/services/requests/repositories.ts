import { handleError } from 'services/response'

import api from 'utils/api'

export const getRepositories = (user: string) => {
  return api
    .get(`users/${user}/repos`)
    .then((response) => response)
    .catch((error) => {
      handleError(error)
    })
}
