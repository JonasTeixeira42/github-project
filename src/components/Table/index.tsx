import React from 'react'

import { Repositories } from 'hooks/use-repositories'
import headers from './header'

import * as S from './styles'

export type TableProps = {
  repositories: Repositories[]
}

const Table = ({ repositories = [] }: TableProps) => {
  return (
    <S.Wrapper role="table">
      <S.Row role="rowgroup">
        {headers.map((header) => (
          <S.Cell key={header} role="cell">
            <S.Header>{header}</S.Header>
          </S.Cell>
        ))}
      </S.Row>
      {repositories.map((repository) => (
        <S.Row key={repository.id} role="rowgroup">
          <S.Cell role="cell">{repository.id}</S.Cell>
          <S.Cell role="cell">{repository.name}</S.Cell>
          <S.Cell role="cell">
            <S.Link href={repository.html_url} target="_blank">
              Go to repository
            </S.Link>
          </S.Cell>
        </S.Row>
      ))}
    </S.Wrapper>
  )
}

export default Table
