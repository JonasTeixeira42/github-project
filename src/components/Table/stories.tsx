import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'

import repositories from './mock'

import Table, { TableProps } from '.'

export default {
  title: 'Table',
  component: Table,
  args: {
    repositories
  }
} as Meta

export const Default: Story<TableProps> = (args) => <Table {...args} />
