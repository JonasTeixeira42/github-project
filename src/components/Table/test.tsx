import React from 'react'
import { screen } from '@testing-library/react'

import { renderWithTheme } from 'utils/tests/helpers'
import repositories from './mock'

import Table from '.'

describe('<Table />', () => {
  it('should render the table correctly', () => {
    const { container } = renderWithTheme(<Table repositories={repositories} />)

    expect(screen.getByRole('heading', { name: /id/i })).toBeInTheDocument()
    expect(
      screen.getByRole('heading', { name: /Repository/i })
    ).toBeInTheDocument()
    expect(screen.getByRole('heading', { name: /url/i })).toBeInTheDocument()

    expect(screen.getByText('1')).toBeInTheDocument()
    expect(screen.getByText('Repo 1')).toBeInTheDocument()
    expect(
      screen.getByRole('link', { name: /Go to repository/i })
    ).toHaveAttribute('href', 'https://www.google.com/')

    expect(container.firstChild).toMatchSnapshot()
  })
})
