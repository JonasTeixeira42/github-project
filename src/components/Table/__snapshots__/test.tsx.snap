// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render the table correctly 1`] = `
.c0 {
  border: 0.1rem solid #f0f0f0;
  border-radius: 0.4rem;
  padding: 0.8rem;
}

.c1 {
  display: grid;
  grid-template-columns: repeat(3,1fr);
}

.c1:not(:last-child) {
  border-bottom: 0.1rem solid #f0f0f0;
  padding-bottom: 0.8rem;
  margin-bottom: 0.8rem;
}

.c2 {
  font-size: 1.4rem;
  padding: 0 0.5rem;
}

.c3 {
  font-size: 1.6rem;
}

<div
  class="c0"
  role="table"
>
  <div
    class="c1"
    role="rowgroup"
  >
    <div
      class="c2"
      role="cell"
    >
      <h2
        class="c3"
      >
        Id
      </h2>
    </div>
    <div
      class="c2"
      role="cell"
    >
      <h2
        class="c3"
      >
        Repository
      </h2>
    </div>
    <div
      class="c2"
      role="cell"
    >
      <h2
        class="c3"
      >
        URL
      </h2>
    </div>
  </div>
  <div
    class="c1"
    role="rowgroup"
  >
    <div
      class="c2"
      role="cell"
    >
      1
    </div>
    <div
      class="c2"
      role="cell"
    >
      Repo 1
    </div>
    <div
      class="c2"
      role="cell"
    >
      <a
        class=""
        href="https://www.google.com/"
        target="_blank"
      >
        Go to repository
      </a>
    </div>
  </div>
</div>
`;
